src-built-in/components/advancedAppLauncher/src/components/LeftNavBottomLinks.jsx

```diff
@@ -1,18 +1,38 @@
 import React from 'react'
 import { getStore } from '../stores/LauncherStore'
 
-
-const bottomEntries = [
-	{ name: "New App", icon: "ff-new-workspace", click: "showAddAppForm" },
-	//{ name: 'New Dashboard', icon: 'ff-dashboard-new' },
-	// { name: "App Catalog", icon: "ff-list", click: "openAppMarket" }
-];
-
 export default class LeftNavBottomLinks extends React.Component {
 
 	constructor(props) {
 		super(props);
+		this.state = {
+			bottomEntries: [
+				{ name: "New App", icon: "ff-new-workspace", click: "showAddAppForm" },
+				//{ name: 'New Dashboard', icon: 'ff-dashboard-new' },
+			]
+		}
+	}
+
+	componentWillMount() {
+		// If user configures the advanced app launcher to use the app catalog, add app catalog button to the bottom entries of the
+		// advanced app launcher menu.
+		finsembleWindow.getOptions((_, opts) => {
+			const useAppCatalog = (((opts || {}).customData || {}).component || {}).useAppCatalog;
```

Comment:
A very confusing and hard to read line. What is the criteria you're looking for?

Reply:
Here I'm using Oliver Steele’s Nested Object Access Pattern. Basically when reading such lines you just ignore the `{}`, `||`, and `()` and just ready it like `opts.customData.component.useAppCatalog`. I didn't use lodash.get here because I don't want to hid the object's path. If you don't think we need to show it we can change to using lodash.get